src/table_builder.cc

```diff
@@ -12,19 +12,47 @@ void TitanTableBuilder::Add(const Slice& key, const Slice& value) {
     return;
   }
 
-  if (ikey.type != kTypeValue || value.size() < cf_options_.min_blob_size) {
-    base_builder_->Add(key, value);
-    return;
-  }
+  if (ikey.type == kTypeBlobIndex &&
+      cf_options_.blob_run_mode == TitanBlobRunMode::kFallback) {
+    assert(vset_ != nullptr);
+
+    // we ingest value from blob file
+    Slice copy = value;
+    BlobIndex index;
+    status_ = index.DecodeFrom(&copy);
+    if (!ok()) {
+      return;
+    }
 
-  std::string index_value;
-  AddBlob(ikey.user_key, value, &index_value);
-  if (!ok()) return;
+    BlobRecord record;
+    PinnableSlice buffer;
+
+    auto storage = vset_->GetBlobStorage(cf_id_).lock();
```

Comment:
instead of have table builder hold the version set, let's have table builder only holding blob storage of the CF. That way this patch don't depend on #13.

Reply:
Still need to pass mutex reference to table factory though.